### ISSUE TITLE
Intermittent failure of twisted.web.test.test_webclient.WebClientSSLTestCase.testTimeoutTriggering on OS X

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2938]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#2938|
|Type|defect|
|Created|2007-12-06 20:46:52Z|
|Branch|https://github.com/twisted/twisted/tree/webclient-timeout-test-2938|

```
[ERROR]: twisted.web.test.test_webclient.WebClientSSLTestCase.testTimeoutTriggering

Traceback (most recent call last):
Failure: twisted.trial.util.DirtyReactorAggregateError: Reactor was unclean.
Selectables:
<HTTPChannel #0 on 60475>
```

<details><summary>Searchable metadata</summary>

```
trac-id__2938 2938
type__defect defect
reporter__exarkun exarkun
priority__high high
milestone__ 
branch__branches_webclient_timeout_test_2938 branches/webclient-timeout-test-2938
branch_author__ 
status__closed closed
resolution__fixed fixed
component__web web
keywords__ 
time__1196974012000000 1196974012000000
changetime__1214145582000000 1214145582000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [23904]) Branching to 'webclient-timeout-test-2938'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun removed owner|
|-|-|

Should be fixed

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve set owner to @exarkun|
|-|-|

Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set status to `closed`|
|-|-|

(In [23975]) Merge webclient-timeout-test-2938

Author: exarkun
Reviewer: therve
Fixes: [#2938](https://github.com/twisted/twisted/issues/2938)

Change a couple `twisted.web.client` tests so that they do not
rely as much on medium-precision timing and so that they clean
up any connections which remain after the test.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
